### PR TITLE
feat: token faucet button

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

...

## Changelog

```
- Please write a list of changes
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

...

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/contexts/developer/actions/requestTokensFromFaucet.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { selectedAccount } from '@core/account'
+import { localize } from '@core/i18n'
+import { showAppNotification } from '@lib/notifications'
+import { get } from 'svelte/store'
+
+export async function requestTokensFromFaucet(): Promise<void> {
+    const url = 'https://faucet.testnet.shimmer.network/api/enqueue'
```
Comment (nicole-obrien) [inline packages/shared/lib/contexts/developer/actions/requestTokensFromFaucet.ts:7]: Move into the network constant

File: packages/shared/routes/dashboard/developerTools/DeveloperTools.svelte [hunk 2]
```diff
@@ -1,32 +1,28 @@
...
                     <Text type="h5" classes="text-left">
-                        {localize('general.nativeTokens')}
+                        {localize('general.assets')}
                     </Text>
-                    <OnboardingButton
-                        primaryText={localize('actions.mintNativeToken')}
-                        secondaryText={localize('general.mintNativeTokenDescription')}
-                        onClick={handleMintNativeTokenClick}
-                    />
+                    <FaucetRequestButton />
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/developerTools/DeveloperTools.svelte:23]: Only show if we have the url available for the network

File: packages/shared/lib/contexts/developer/actions/requestTokensFromFaucet.ts (hunk 1 above)
Comment (nicole-obrien) [inline packages/shared/lib/contexts/developer/actions/requestTokensFromFaucet.ts:7]: Move into the network constant

**Review by nicole-obrien — COMMENTED** [review]
Some small changes I need to make

File: packages/shared/components/popups/ConfirmationPopup.svelte [hunk 3]
```diff
@@ -13,24 +13,32 @@
...
+        closePopup()
     }
 
     function cancelClick(): void {
         if (onCancel) {
             onCancel()
-        } else {
-            closePopup()
         }
+        closePopup()
```
Comment (begonaalvarezd) [inline packages/shared/components/popups/ConfirmationPopup.svelte:26]: I think this change here collides with the current behavior of async activities `packages/shared/components/popups/ActivityDetailsPopup.svelte#59`. Async tx > open detail > click reject > cancel > popups should be kept open but on the previous step

File: packages/shared/components/TextHint.svelte [hunk 4]
```diff
@@ -26,19 +26,19 @@
...
             _icon = 'success-error'
             _iconClasses = 'text-red-500 dark:text-red-500'
         } else if (warning) {
-            _classes = 'bg-yellow-50 dark:bg-gray-800'
+            _classes = 'bg-yellow-50 dark:bg-yellow-50 border-yellow-500'
             _icon = 'exclamation'
             _iconClasses = 'text-yellow-500 dark:text-yellow-500'
         } else if (info) {
-            _classes = 'bg-blue-50 dark:bg-gray-800'
+            _classes = 'bg-blue-50 dark:bg-blue-50 border-blue-500'
```
Comment (begonaalvarezd) [inline packages/shared/components/TextHint.svelte:41]: in darkmode the texthints are a bit light, maybe something we could improve by adding bg-opacity in darkmode, but that would also require changing the text color. Just a note

Comment (nicole-obrien) [reply packages/shared/components/TextHint.svelte:41]: we are waiting on designs for the text hint this is okay for now

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Added 2 comments, the darkmode style one isnot important, just something to note. 
The other other is about about how moving `closePopup` out of the `else` affects the current flow of async tx > open details > reject > cancel 

Apart from that, the PR looks and works great 🚀 Amazing work

**Review by begonaalvarezd — APPROVED** [review]
💎 works!

**Pull request merged**